network-store-model/src/main/java/com/powsybl/network/store/model/ExtensionLoaders.java

```diff
@@ -23,15 +24,43 @@ public final class ExtensionLoaders {
 
     private static final ServiceLoaderCache<ExtensionLoader> EXTENSION_LOADERS = new ServiceLoaderCache<>(ExtensionLoader.class);
 
+    private static Predicate<ExtensionLoader> namePredicate(String name) {
+        return s -> s.getName() != null && name.equals(s.getName());
+    }
+
+    private static <K extends Extension> Predicate<ExtensionLoader> typePredicate(Class<? super K> type) {
+        return s -> type == s.getType();
+    }
+
+    private static <K extends ExtensionAttributes> Predicate<ExtensionLoader> attributesTypePredicate(Class<? super K> attributesType) {
+        return s -> attributesType.isAssignableFrom(s.getAttributesType());
+    }
+
     private ExtensionLoaders() {
     }
 
-    public static ExtensionLoader findLoader(String name) {
-        return findLoader(s -> s.getName() != null && name.equals(s.getName()), name);
+    public static boolean loaderExists(String name) {
+        return loaderExists(namePredicate(name));
+    }
+
+    public static <K extends Extension> boolean loaderExists(Class<? super K> type) {
+        return loaderExists(typePredicate(type));
+    }
+
+    private static boolean loaderExists(Predicate<ExtensionLoader> predicate) {
+        return EXTENSION_LOADERS.getServices().stream().anyMatch(predicate);
+    }
+
+    public static ExtensionLoader findLoaderByName(String name) {
+        return findLoader(namePredicate(name), name);
+    }
+
+    public static <K extends Extension> ExtensionLoader findLoader(Class<? super K> type) {
```

Comment:
before, findLoader was by name but now by type ? Better to make it explicit findLoaderByType ?

Reply:
I wanted to make it consistent with the powsybl API where we have getExtension(Class<? super E> type); and getExtensionByName(String name);
https://github.com/powsybl/powsybl-core/blob/05311a464ed32c1ae83d4bac76d00a367eb3d9a8/commons/src/main/java/com/powsybl/commons/extensions/Extendable.java#L26-L41
I'm also ok with findLoaderByType which is indeed clearer but not consistent with the getExtension api.